flink-runtime/src/test/java/org/apache/flink/runtime/source/coordinator/SourceCoordinatorAlignmentTest.java

```diff
@@ -310,6 +310,53 @@ void testWatermarkAggregatorRandomly() {
         testWatermarkAggregatorRandomly(10, 10000, true, false);
     }
 
+    @Test
+    void testWatermarkAlignmentWhileSubtaskFinished() throws Exception {
+        long maxDrift = 1000L;
+        WatermarkAlignmentParams params =
+                new WatermarkAlignmentParams(maxDrift, "group1", maxDrift);
+
+        final Source<Integer, MockSourceSplit, Set<MockSourceSplit>> mockSource =
+                createMockSource();
+
+        sourceCoordinator =
+                new SourceCoordinator<MockSourceSplit, Set<MockSourceSplit>>(
+                        OPERATOR_NAME,
+                        mockSource,
+                        getNewSourceCoordinatorContext(),
+                        new CoordinatorStoreImpl(),
+                        params,
+                        null) {
+                    @Override
+                    void announceCombinedWatermark() {
+                        super.announceCombinedWatermark();
+                    }
+                };
+
+        sourceCoordinator.start();
+
+        int subtask0 = 0;
+        int subtask1 = 1;
+
+        setReaderTaskReady(sourceCoordinator, subtask0, 0);
+        setReaderTaskReady(sourceCoordinator, subtask1, 0);
+        registerReader(subtask0);
+        registerReader(subtask1);
+
+        reportWatermarkEvent(sourceCoordinator, subtask0, 42);
+        assertLatestWatermarkAlignmentEvent(subtask0, 1042);
+
+        reportWatermarkEvent(sourceCoordinator, subtask1, 44);
+        assertLatestWatermarkAlignmentEvent(subtask1, 1042);
+
+        // mock noMoreSplits event
+        assertHasNoMoreSplits(subtask0, true);
```

Comment:
I'm not sure what `testWatermarkAlignmentWhileSubtaskFinished` wants to test?

IIUC, the test still works even if we don't call `assertHasNoMoreSplits(subtask0, true);` (remove this line) here, right?

Reply:
> My intention is to add a test case to simulate the sending of the noMoreSplits event

Actually, this test is testing the watermark result when the watermark of one subtask is Long.Max, right?

I think this test can be removed.